### DISCUSSION TITLE
Counsel Clarification

### DISCUSSION THREAD

**Discussion by alanfermtz** [AntonyCorbett/OnlyT#475] [Q&A] [answered]
I am having trouble to understand how the program knows and set how many time the chairman Counsel and Advice works, especially between last Apply Yourself and the mid song.

I have been doing test meeting to understand the program, but I have seen inconsistencies in this particular part.

Idk if I miss a wiki related to this or any discussion before

Thanks

**Comment by AntonyCorbett** [comment]
See in the wiki - **Automatic mode => Persist student time** - Check this box if you want to display a student's talk duration after the conclusion of the talk. This can be useful for the chairman. The display shows how long the student talk took. This static display remains on-screen for a little over 1 minute so that the chairman can refer to it if needed. When the minute is up, the timer display reverts to the standard clock - giving the chairman notice that his allotted time is up. (the chairman needs to understand these cues.)

**Reply by alanfermtz** [reply]
Thanks for the response, I understand that. But I mean in the timing report

**Comment by AntonyCorbett** [comment, marked as answer]
@alanfermtz Regarding the duration of counsel as displayed in the Timing Report, this is calculated in the PdfTimingReport.cs code file in the OnlyT.Report assembly. The algorithm is summarised below:

For most items in the Ministry section of the meeting, the duration between the end of the item and the start of the next item (assuming both are recorded by OnlyT) is the starting point. 20 secs is deducted to allow for the 2 changeovers between the speakers and chairman. The value is then rounded up to the nearest 5 seconds. 

So, for example, if the Bible Reading ends at 19:31:23 and the first Ministry Talk starts at 19:33:04, the interim duration is 101 secs. We deduct 20 secs to give us 81 secs. This is rounded up to the nearest 5 secs (85 secs) which is considered 25 secs overtime (the algorithm assumes 1 minute should be used for counsel).

For the final item in this section of the meeting, OnlyT has no timing data for the start of the next "item" (i.e. the middle song). In this case, 3 mins 20 secs is allotted for the song (or "Interim Segment" as it appears in the report) and the counsel duration is deduced on this basis along with the start time of the first item in the "Living" section.

So, for example, if the final Ministry talk ends at 19:47:58 and the first Living item starts at 19:52:58, we work backwards deducting 3 mins 20 secs for the song, 10 secs for speaker changeover at the end of the item and a further 15 secs for speaker changeover at the end of the section. Taking 3 mins 45 secs from 19:52:58 we arrive at 19:49:13. So we estimate the counsel duration to be 75 secs (19:49:13 - 19:47:58).

Of course, the algorithm provides a rough-and-ready approach to the counsel duration because the actual duration is not explicitly timed.

Hope this helps. And of course, I'd welcome any thoughts on improving the algorithm.
